app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -161,6 +165,13 @@ constructor(
             followedTagFilter = getTagsAndSubTags(_followedTags.value).toList()
             sectionTags = repository.getSubTags(sectionTagId)
             semesterTags = repository.getSubTags(semesterTagId)
+            val pictureByteArray = repository.getUserProfilePicture(userId)
+            _profilePicture.value =
+                if (pictureByteArray != null) {
+                    BitmapFactory.decodeByteArray(pictureByteArray, 0, pictureByteArray.size)
+                } else {
+                    null
```

Comment:
Is the else block necessary? The default value is set to null, so you could safely get rid of the else block.